src/vs/workbench/parts/git/common/git.ts

```diff
@@ -339,7 +339,11 @@ export interface IAskpassService {
 }
 
 // Utils
+const invalidBranchPatternName = /(^\.)|(\/\.)|(\.\.)|(~)|(\^)|(:)|(\/$)|(\.lock$)|(\.lock\/)|(\\)|(\*)|(\s)|(^\s*$)|(\.$)/g;
```

Comment:
Why wrapping everything in groups? The regex is now broken since the `^` and `$` were making sure the whole string is matched.